commandapi-nms/commandapi-1.19-common/src/main/java/dev/jorel/commandapi/nms/NMS_1_19_Common.java

```diff
@@ -577,6 +536,111 @@ public final ParticleData<?> getParticle(CommandContext<CommandSourceStack> cmdC
+		final Color color = Color.fromRGB((int) (options.getColor().x() * 255.0F),
+			(int) (options.getColor().y() * 255.0F), (int) (options.getColor().z() * 255.0F));
+		final Color toColor = Color.fromRGB((int) (options.getToColor().x() * 255.0F),
+			(int) (options.getToColor().y() * 255.0F), (int) (options.getToColor().z() * 255.0F));
+		return new ParticleData<>(particle, new DustTransition(color, toColor, options.getScale()));
+	}
+
+	/**
+	 * {@link #getParticle(CommandContext, String)}  Parent Method}
+	 * @param particle
+	 * @param options
+	 * @return
+	 */
+	private ParticleData<DustOptions> getParticleDataAsDustParticleOptions(Particle particle, DustParticleOptions options) {
+		final Color color = Color.fromRGB((int) (options.getColor().x() * 255.0F),
+			(int) (options.getColor().y() * 255.0F), (int) (options.getColor().z() * 255.0F));
+		return new ParticleData<DustOptions>(particle, new DustOptions(color, options.getScale()));
+	}
+
+	/**
+	 * {@link #getParticleDataAsVibrationParticleOption(CommandContext, Level, Particle, VibrationParticleOption)
+	 * Parent Method}
+	 * @param
+	 * @param
+	 * @return
+	 */
+	private ParticleData<?> getParticleDataAsVibrationParticleOption(CommandContext<CommandSourceStack> cmdCtx,
+	                                                                 Level level, Particle particle,
+	                                                                 VibrationParticleOption options) {
+		// The "from" part of the Vibration object in Bukkit is completely ignored now,
+		// so we just populate it with some "feasible" information
+		final Vec3 origin = cmdCtx.getSource().getPosition();
+		Location from = new Location(level.getWorld(), origin.x, origin.y, origin.z);
+
+		final Destination destination;
+
+		if (options.getDestination() instanceof BlockPositionSource positionSource) {
+			destination = getVibrationParticleOptionDestinationAsBlockPositionSource(positionSource, level);
+		} else if (options.getDestination() instanceof EntityPositionSource positionSource) {
+			destination = getVibrationParticleOptionDestinationAsEntityPositionSource(positionSource, level);
+		} else {
+			CommandAPI.getLogger().warning("Unknown vibration destination " + options.getDestination());
+			return new ParticleData<Void>(particle, null);
+		}
+		return new ParticleData<Vibration>(particle, new Vibration(from, destination, options.getArrivalInTicks()));
+	}
+	
+	/**
+	 * {@link #getParticleDataAsVibrationParticleOption(CommandContext, Level, Particle, VibrationParticleOption)
+	 * Parent Method}
+	 * @param positionSource {@link BlockPositionSource} the block position source to find the designation of.
+	 * @param level {@link Level} The level in which to find the destination
+	 * @return {@link Destination} the destination of the provided source in the given level.
+	 */
+	private @NotNull Destination getVibrationParticleOptionDestinationAsBlockPositionSource(@NotNull BlockPositionSource positionSource, Level level) {
```

Comment:
This method is really small, did it really need its own method for these two statements?